File: 494-target-sum/target-sum.cpp
class Solution {
public:
    int findTargetSumWays(vector<int>& nums, int target) {
        vector<int> curr;
        curr.push_back(0);

        for (int i = 0; i < nums.size(); i++) {
            vector<int> next(curr.size() * 2);
            int si = 0;

            for (int val : curr) {
                next[si++] = val + nums[i];
                next[si++] = val - nums[i];
            }

            curr = next;
        }

        int ans = 0;
        for (int val : curr)
            if (val == target)
                ans++;

        return ans;
    }
};

File: 494-target-sum/target-sum.java
class Solution {
    public int findTargetSumWays(int[] nums, int target) {
        return cal(nums,0,0,target);
    }
    public int cal(int[] nums, int index, int sum, int target){
        if(index == nums.length) return sum == target? 1 : 0;
        int plus = cal(nums, index + 1, sum + nums[index], target);
        int minus = cal(nums, index + 1, sum - nums[index], target);
        return plus+minus;
    }
}